contrib/api_incubator/src/main/java/io/opentelemetry/trace/SpanEx.java

```diff
@@ -0,0 +1,201 @@
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.trace.Span.Kind;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SpanEx {
+
+  private final Span span;
+
+  public SpanEx(Span span) {
+    this.span = span;
+  }
+
+  Span getSpan() {
+    return span;
+  }
+
+  public void setStatus(Status status) {
+    span.setStatus(status);
+  }
+
+  public void updateName(String name) {
+    span.updateName(name);
+  }
+
+  public SpanContext getContext() {
+    return span.getContext();
+  }
+
+  public boolean isRecording() {
+    return span.isRecording();
+  }
+
+  public void end() {
+    span.end();
+  }
+
+  public void end(EndSpanOptions endOptions) {
+    span.end(endOptions);
+  }
+
+  public static Builder newBuilder(Tracer tracer, String spanName) {
+    return new Builder(tracer, spanName);
+  }
+
+  public static class Builder {
+
+    private final Span.Builder builder;
+
+    public Builder(Tracer tracer, String spanName) {
+      builder = tracer.spanBuilder(spanName);
+    }
+
+    public SpanEx.Builder setParent(SpanEx parent) {
+      builder.setParent(parent.span);
+      return this;
+    }
+
+    public SpanEx.Builder setParent(SpanContext remoteParent) {
+      builder.setParent(remoteParent);
+      return this;
+    }
+
+    public SpanEx.Builder setNoParent() {
+      builder.setNoParent();
+      return this;
+    }
+
+    // option 1... an interface that encapsulates the attribute nature
+    public SpanEx.Builder addLink(SpanContext context, Attributes attributes) {
+      builder.addLink(context, makeAttributeMap(attributes));
+      return this;
+    }
+
+    // option 2... a type-safe key-value pair with varargs parameters
+    public SpanEx.Builder addLink(SpanContext context, Attribute... attributes) {
+      builder.addLink(context, makeAttributeMap(attributes));
+      return this;
+    }
+
+    // option 3... provide a concrete Link class.
+    public SpanEx.Builder addLink(LinkEx link) {
+      return addLink(link.getSpanContext(), link.getAttributes());
+    }
+
+    public SpanEx.Builder setSpanKind(Kind spanKind) {
+      builder.setSpanKind(spanKind);
+      return this;
+    }
+
+    public SpanEx.Builder setStartTimestamp(long startTimestamp) {
+      builder.setStartTimestamp(startTimestamp);
+      return this;
+    }
+
+    public SpanEx startSpan() {
+      return new SpanEx(builder.startSpan());
+    }
+
+    // option 1... provide a single attribute
+    public SpanEx.Builder setAttribute(Attribute attribute) {
+      builder.setAttribute(attribute.key().key(), makeValue(attribute));
+      return this;
+    }
+
+    // option 2... a type-safe key-value pair with varargs parameters
+
+    /** doc me. */
+    public SpanEx.Builder setAttribute(Attribute... attributes) {
+      for (Attribute attribute : attributes) {
+        builder.setAttribute(attribute.key().key(), makeValue(attribute));
+      }
+      return this;
+    }
```

Comment:
comparing

```
Attribute attr1 = Attribute.create(key1, value1);
Attribute attr2 = Attribute.create(key2, value2);
Attribute attr3 = Attribute.create(key3, value3);
spanBuilder.setAttribute(attr1, attr2, attr3);
```

and

```
spanBuilder.setAttribute(key1, value1)
           .setAttribute(key2, value2)
           .setAttribute(key3, value3);
```

the latter is more efficient, and seems more user-friendly to me

Reply:
check out the latest updates, which have support for this usage.